Quick refresh of the Repository Management doc

designsimply opened the pull request:
This PR seeks to update the Repository Management documentation to reflect recent practices including the following:

- Reorganize the labels section.
- General updates to the triage section.
- Add a note about how to handle help requests.
- Update common examples of labels, milestones, and projects.
- Change 'backlog' to 'list'.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows the accessibility standards.

designsimply committed:
Quick refresh of the Repository Management doc

This PR seeks to update the Repository Management documentation to reflect recent practices including the following:

- Reorganize the labels section.
- General updates to the triage section.
- Add a note about how to handle help requests.
- Update common examples of labels, milestones, and projects.
- Change 'backlog' to 'list'.

youknowriad commented on docs/contributors/repository-management.md:
```diff
@@ -32,53 +40,39 @@ To better organize the issue backlog, all issues should have [one or more labels
+Here are some milestones you might see:
+
+- [WordPress 5.x](https://github.com/WordPress/gutenberg/milestone/70): Tasks that should be done for future WordPress releases such as 5.1 or 5.2.
+- [5.0 (Gutenberg)](https://github.com/WordPress/gutenberg/milestone/85): PRs targeted for the Gutenberg Plugin 5.0 release.
```
I wonder if we should be specific about the versions here, as it will force us to update this document regularly.

designsimply replied:
Can totally make versions less specific, however, I also don't think older ones need to be updated regularly (they were 2.2 and 2.3 before) to be useful. I only updated them because it seemed to make sense as part of a refresh. In any case, I will make them less specific. Thank you!

youknowriad commented on docs/contributors/repository-management.md:
```diff
@@ -32,53 +40,39 @@ To better organize the issue backlog, all issues should have [one or more labels
 - If you feel comfortable triaging the issue further, then you can also:
   - Check that the bug report is valid by debugging it to see if you can track down the technical specifics.
-  - Check if the issue is missing some detail and see if you can fill in those details. For instance, if a bug report is missing visual detail, it’s helpful to reproduce the issue locally and upload a GIF.
+  - Check if the issue is missing some detail and see if you can fill in those details. For instance, if a bug report is missing visual detail, it’s helpful to reproduce the issue locally and upload a screenshot or GIF.
```
Can we add some rules like.

 - Close the issue if it has a "Needs More Info" label but the author didn't respond in 2 weeks.
 - Remove "Needs More Info" label if the author of the issue responded and have enough details.

designsimply replied:
Yes! Adding!

youknowriad commented:
Thanks, this is great 👍

youknowriad commented:
Should we mention "Good First Issue" and "Good First Review" labels somewhere?

gziolo commented:
Yes, please include `Good First Review`  👍

designsimply committed:
Quick refresh of the Repository Management doc

Updating based on feedback received: 

- Be less specific about the version numbers in example milestones.
- Add more guidelines about the Needs More Info label.
- Mention "Good First Issue" and "Good First Review" labels.

gziolo committed:
Update repository-management.md

gziolo commented on docs/contributors/repository-management.md:
```diff
@@ -69,6 +71,8 @@ When reviewing issues, here are some steps you can perform:
 - If the title doesn’t communicate the issue, edit it for clarity (requires proper permissions).
 - If it’s a bug report, test to confirm the report or add the `Needs Testing` label. If there is not enough information to confirm the report, add the `[Status] Needs More Info` label and ask for the details needed.
+- Remove the `[Status] Needs More Info` if the author of the issue has responded with enough details.
+- Close the issue with a note if it has a `[Status] Needs More Info` label but the author didn't respond in 2+ weeks.
```
👍

gziolo reviewed (approved):
Looks good speaking myself. I would defer the final 👍 based on opinions from @youknowriad, @mapk and @desrosj.

youknowriad reviewed (approved):
Let's get this in and iterate.

gziolo commented:
I merged this one. I would love still to see feedback on this one. We can always open a follow-up PR :)
